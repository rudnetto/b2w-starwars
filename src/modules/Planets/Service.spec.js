describe('Planets service test suite', () => {
  it('should retrieve data correctly', () => {
    const httpResultMock = {
      data: {
        name: 'Tatooine',
        population: '120000',
        terrain: 'Dessert',
        climate: 'Arid',
        films: [
          'episodeI',
          'episodeIII',
          'episodeIV',
          'episodeV',
          'episodeVI',
        ],
      },
    };
    const get = jest.fn(() => httpResultMock);
    jest.doMock('../../utils/HttpClient', () => ({ get }));
    jest.doMock('../../utils/Random', () => ({ generateRandomNumber: () => 1 }));

    const Service = require('./Service').default; // eslint-disable-line
    const result = Service.getRandomPlanet();

    expect(get).toBeCalledWith('/planets/1');
    expect(result).toMatchSnapshot();
  });
});
