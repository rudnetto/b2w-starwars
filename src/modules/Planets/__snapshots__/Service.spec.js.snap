// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets service test suite should retrieve data correctly 1`] = `Promise {}`;
